apps/vc-api/src/db-config.ts

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright 2021 - 2023 Energy Web Foundation
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { TypeOrmModule } from '@nestjs/typeorm';
+import { homedir } from 'os';
+
+/**
+ * Inspired by https://dev.to/webeleon/unit-testing-nestjs-with-typeorm-in-memory-l6m
+ * @param inMemory // default set to false; set to true for unit testing  
+ */
+export const TypeOrmSQLiteModule = (inMemory=false) =>
+  TypeOrmModule.forRoot({
+    type: 'sqlite',
+    database: !inMemory ? 
+      `${homedir}/.afj/data/wallet/${process.env['ASKAR_WALLET_ID']}/sqlite.db` // path to askar wallet db
```

Comment:
Hey @ShikharBhatt , thanks for the PR. It's great to see how we can set the path for database.

However, I'm skeptical that we should try to use the same database for exchanges data as for Credo managed data. I think it is adding coupling that isn't really necessary. For instance, we're coupled in the db type (as this PR shows) and it could complicate data migrations in the future. 

I think my ideal design would be that we put the Credo/AFJ database in the same directory as the "exchanges" db, but my thinking is that having them in the `.afj` directory doesn't really make sense and having a shared `db` directory would be preferable. However, perhaps this is a bit too much effort for now.

I propose we have `joi` config like this:
```
exchangesDatabase: Joi.object({
    inMemory: Joi.boolean().default(true),
    path: Joi.string().when('inMemory', {
      is: false,
      then: Joi.required(),
      otherwise: Joi.optional()
    })
  }).default(() => ({}))
```
This should make the default config as follows I believe
```
exchangesDatabase: {
    inMemory: true
  }
```
Note that the configService should be used to access this (not `process.env`) see https://docs.nestjs.com/techniques/configuration

Reply:
> We can create a common db path - ${homedir}/.vc-api

Good idea 👍 

Re the joi config, I can see why it might have been confusing as joi supports nested config properties but env files don't my initial idea wouldn't easily work. We could switch to yaml, but that's too much for now IMO.
Flatting my proposal so that it works with .env, we could change the object in this file https://github.com/impactility/vc-api/blob/develop/apps/vc-api/src/config/env-vars-validation-schema.ts would become something like
```typescript
export const envVarsValidationSchema = Joi.object({
  NODE_ENV: Joi.string().valid('development', 'production', 'test').default('development'),
  PORT: Joi.number().integer().positive().default(3000),
  BASE_URL: Joi.string().uri().default('http://localhost:3000'),
  EXCHANGES_DB_IN_MEMORY: Joi.boolean().default(true),
  EXCHANGES_DB_PATH: Joi.string().when('EXCHANGES_DB_IN_MEMORY', {
      is: false,
      then: Joi.required(),
      otherwise: Joi.optional()
    })
  })
});
```
What do you think?

Reply:
@ShikharBhatt , to confirm my understanding from our call on Friday, what I remember is:
- We won't go with the joi config shown above. It's a bit too complicated for now.
- The approach that you showed in our call looked good to me. I suppose this PR just need to be updated with your latest code.

Reply:
@jrhender Yes, I will update the PR in a few minutes and it'll be ready for review.

Reply:
@jrhender I've made the necessary changes.
- added a`Joi` configuration for `DB_BASE_PATH`
- updated the db paths respectively for credo and exchanges
- updated the `TypeOrmSQLiteModule` to consume `ConfigService`

Let me know if these changes make sense.
Thanks